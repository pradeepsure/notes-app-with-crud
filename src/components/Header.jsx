import React, {Component} from 'react';
import PropTypes from 'prop-types';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import { withStyles } from '@material-ui/core/styles';
import AssignmentIcon from '@material-ui/icons/Assignment';
import Typography from '@material-ui/core/Typography';
import SearchBar from './SearchBar';
import { Button } from '@material-ui/core';
import {firebase, googleAuthProvider} from '../firebase/firebase';


const styles = theme => ({
  root: {
    width: '100%',
  },
  grow: {
    flexGrow:1
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
  title: {
    display: 'none',
    [theme.breakpoints.up('sm')]: {
      display: 'block',
    },
  },
});

class Header extends Component {
  constructor(props){
    super(props);
    this.state = {
      isLoggedIn: localStorage.getItem('isLoggedIn') || false
    }; 
    this.handleLogin=this.handleLogin.bind(this);
    this.handleLogOff=this.handleLogOff.bind(this); 
    this.onToggleChange = this.onToggleChange.bind(this);  
  }  
  onToggleChange() {
    const changeView = this.props.viewType === 'gridView' ? 'listView' : 'gridView';
    this.props.toggleView(changeView);
  }

  handleLogin(){
    return firebase.auth().signInWithPopup(googleAuthProvider)
    .then(()=>{
      this.setState({ isLoggedIn: true });
      localStorage.setItem('isLoggedIn', true);
    });
  }

  handleLogOff(){
    return firebase.auth().signOut()
    .then(()=>{
      this.setState({ isLoggedIn: false });
      localStorage.removeItem('isLoggedIn');
    })
  }

  render(){
    const { classes, viewType } = this.props;
    return (
      <div className={classes.root}>
        <AppBar position="static">
          <Toolbar>
          <AssignmentIcon />
          <Typography className={classes.title} variant="h6" color="inherit" noWrap>
            Keep - Notes
          </Typography>  
            {(this.state.isLoggedIn)?<SearchBar />: ''}       
            <div className={classes.grow}></div>
              <Button onClick={this.state.isLoggedIn ? this.handleLogOff: this.handleLogin}>
               {this.state.isLoggedIn ? 'SignOff' : 'Login'  }
              </Button>
              {/* <Button variant="contained" color="secondary" className={classes.button} onClick={this.onToggleChange}>
                {viewType === 'gridView' ? 'List' : 'Grid'}
              </Button> */}
          </Toolbar>
        </AppBar>
      </div>
    );
  }


}

Header.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(Header);    // HOC -> Higher Order Component which takes another compoennt as an argument
