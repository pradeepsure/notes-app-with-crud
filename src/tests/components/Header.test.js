import Header from '../../components/Header';
import React from 'react';
import { shallow, mount } from 'enzyme';
import { createSerializer } from 'enzyme-to-json';

expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));

let wrapper;
beforeEach(() => {    
    wrapper = shallow(
        <Header/>
    );
});


test('Testing Header component.... ', () => {        
    expect(wrapper).toMatchSnapshot();
});