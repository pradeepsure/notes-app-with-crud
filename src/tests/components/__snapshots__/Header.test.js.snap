// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header component....  1`] = `
<Header
  classes={
    Object {
      "grow": "Header-grow-2",
      "menuButton": "Header-menuButton-3",
      "root": "Header-root-1",
      "title": "Header-title-4",
    }
  }
/>
`;
